ethpm_cli/package.py

```diff
@@ -0,0 +1,120 @@
+from collections import namedtuple
+import json
+from typing import Any, Dict, Iterable, Tuple  # noqa: F401
+from urllib import parse
+
+from eth_utils import to_dict, to_text
+from ethpm.backends.http import GithubOverHTTPSBackend
+from ethpm.backends.ipfs import BaseIPFSBackend
+from ethpm.backends.registry import RegistryURIBackend
+from ethpm.typing import Address, Manifest, URI  # noqa: F401
+from ethpm.utils.ipfs import extract_ipfs_path_from_uri, generate_file_hash, is_ipfs_uri
+from ethpm.utils.manifest_validation import (
+    validate_manifest_against_schema,
+    validate_manifest_deployments,
+    validate_raw_manifest_format,
+)
+from ethpm.utils.uri import (
+    is_valid_content_addressed_github_uri,
+    parse_registry_uri,
+    validate_blob_uri_contents,
+)
+from ethpm.validation import is_valid_registry_uri, validate_package_name
+
+from ethpm_cli._utils.ipfs import get_ipfs_backend
+from ethpm_cli.exceptions import UriNotSupportedError
+
+
+class Package:
+    def __init__(self, target_uri: URI, alias: str = None, ipfs: bool = None) -> None:
+        self.ipfs = get_ipfs_backend(ipfs)
+        resolved_target_uri = resolve_target_uri(target_uri)
+        self.manifest_uri: URI = resolved_target_uri.manifest_uri
+        self.registry_address: Address = resolved_target_uri.registry_address
+
+        resolved_manifest_uri = resolve_manifest_uri(self.manifest_uri, self.ipfs)
+        self.raw_manifest: bytes = resolved_manifest_uri.raw_manifest
+        self.resolved_content_hash: str = resolved_manifest_uri.resolved_content_hash
+
+        self.manifest: Manifest = process_and_validate_raw_manifest(self.raw_manifest)
+        self.alias = determine_alias(self.manifest["package_name"], alias)
+        self.target_uri = target_uri
+
+    @to_dict
+    def generate_ethpm_lock(self) -> Iterable[Tuple[str, str]]:
+        yield "resolved_uri", self.manifest_uri
+        yield "resolved_content_hash", self.resolved_content_hash
+        yield "target_uri", self.target_uri
+        yield "registry_address", self.registry_address
+        yield "alias", self.alias
+        yield "resolved_version", self.manifest["version"]
+        yield "resolved_package_name", self.manifest["package_name"]
+
+
+def determine_alias(pkg_name: str, alias: str = None) -> str:
+    if alias:
+        validate_package_name(alias)
```

Comment:
I think this validation is happening at the wrong level.  At this stage it should probably be assumed to be valid with validation having occurred higher up (like in the `Config` object, or as soon as the CLI args are parsed and you know what the alias is).